pkg/ingester/circuitbreaker.go

```diff
@@ -188,17 +188,28 @@ func (cb *circuitBreaker) isActive() bool {
 }
 
 func (cb *circuitBreaker) activate() {
-	if cb == nil {
+	if cb == nil || cb.active.Load() {
 		return
 	}
 	if cb.cfg.InitialDelay == 0 {
+		level.Info(cb.logger).Log("msg", "activating circuit breaker", "requestType", cb.requestType)
 		cb.active.Store(true)
+		return
 	}
 	time.AfterFunc(cb.cfg.InitialDelay, func() {
+		level.Info(cb.logger).Log("msg", "activating circuit breaker", "requestType", cb.requestType)
 		cb.active.Store(true)
 	})
 }
 
+func (cb *circuitBreaker) deactivate() {
```

Comment:
So this PR basically adds a support to deactivate circuit breakers, but it doesn't actually deactivate them anywhere?

Reply:
Correct -- I wanted to keep the changes small, so will add the invocation in another PR. The plan is to call it from https://github.com/grafana/mimir/blob/main/pkg/ingester/downscale.go when we change the read-only state.